PocketKit/Sources/Sync/RemoteMapping/SavedItem+RemoteMapping.swift

```diff
@@ -40,14 +40,12 @@ extension SavedItem {
         }
         
         remote.tags?.compactMap { $0 }.forEach({ remoteTag in
-            let fetchRequest = Requests.fetchTag(byName: remoteTag.name)
-            fetchRequest.fetchLimit = 1
-            let tag = try? context.fetch(fetchRequest).first ?? Tag(context: context)
-            guard let tag = tag else { return }
-            tag.update(remote: remoteTag)
+            let tag = space.fetchOrCreateTag(byName: remoteTag.name)
             addToTags(tag)
         })
         
+        try? space.deleteOrphanTags()
```

Comment:
@jacobsimeon as discussed created a ticket for backlog, let me know if I missed any details 
https://getpocket.atlassian.net/browse/IN-788